src/supa/main.py

```diff
@@ -105,9 +107,34 @@ def callback(ctx: Context, param: Option, value: Optional[str]) -> Optional[str]
     )(f)
 
 
+def log_level_option(f):  # type: ignore
+    """Define common option for specifying log level."""
+
+    def callback(ctx: Context, param: Option, value: Optional[str]) -> Optional[str]:
+        """Update the Settings instance when the database-file option is used."""
+        cos: CommonOptionsState = ctx.ensure_object(CommonOptionsState)
+        if value in ("DEBUG", "INFO", "WARNING", "ERROR", "CRITICAL"):
+            cos.log_level = value
+            settings.log_level = cos.log_level
+
+            # not certain if this is the best way of setting the log level, but it does seem to do the job
+            logging.root.manager.root.setLevel(cos.log_level)
+        return value
+
+    return click.option(
+        "--log-level",
+        type=str,
+        expose_value=False,  # Don't add to sub command arg list. We have `@pass_common_options_state` for that.
+        help="Log level (DEBUG, INFO, WARNING, ERROR, CRITICAL)",
```

Comment:
It could be changed into a `click.Choice` option so that the loglevel choice is validated

https://click.palletsprojects.com/en/8.1.x/options/#choice-options